dashersw/cote
# [Monitoring Tool][WIP] Implement interactive monitoring tool

## What

This PR implements a monitoring tool for the cote micro service architecture.
Currently, the architecture is implemented with D3v4.
## TODO
- [x] Make the micro service topology visible with D3 force graph.
- [ ] Make the link lengths dynamically configurable.
- [ ] Add a table view to display further information about the node, process or host.
- [ ] Feed the output of the console monitor to this page.
- [ ] Add a xy-graph to show the CPU usage of the processes, nodes.
## Test
- Checkout this branch.
- Run `examples/monitor.js` in your terminal to start the monitoring tool.
- Open `localhost:5555` in your browser to see the monitoring tool.
- Run `node examples/responder.js` to add more processes and nodes.
- To test it with a more crowded topology run `node examples/requester.js`
